server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -410,6 +420,18 @@ void requestStop() {
     serverStopRequested = true;
   }
 
+  public ThreadPoolExecutor getWriteThreadExecutor() {
+    var configured =
+        getServerConfig().getConfiguration().getCount(Property.TSERV_WRITE_THREADS_MAX);
+    // resize pool if config changed
+    if (INITIAL_MAX_WRITE_THREADS != configured) {
+      if (configured == 0) {
+        return writeThreadExecutor;
+      }
+    }
+    return writeThreadExecutor;
```

Comment:
This doesn't look like it's doing any resizing at all. Is this a follow-on action? I think we could use the deriver stuff in the configuration to detect when resize is needed, rather than trying to save the initial configuration and watch it manually. I know we have another thread pool that is dynamically resized. This could be done however that one is being done.